src/test/regress/sql/pg16.sql

```diff
@@ -650,10 +622,80 @@ RESET citus.enable_ddl_propagation;
 -- by default, admin option is false, inherit is true, set is true
 GRANT role3 TO role4;
 GRANT role3 TO role5 WITH ADMIN TRUE, INHERIT FALSE, SET FALSE;
-SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option, set_option FROM pg_auth_members WHERE roleid::regrole::text = 'role3' ORDER BY 1, 2;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option, set_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role3' ORDER BY 1, 2;
 
 DROP ROLE role3, role4, role5;
 
+-- Test that everything works fine for roles that are propagated
+-- Grant role with admin and inherit options set to true
+SET citus.enable_ddl_propagation TO on;
+CREATE ROLE role6;
+CREATE ROLE role7;
+GRANT role6 TO role7 WITH ADMIN OPTION, INHERIT TRUE;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role6' ORDER BY 1, 2;
+
+-- GRANT with INHERIT and SET Options
+GRANT role6 TO role7 WITH INHERIT TRUE, SET TRUE;
+SELECT roleid::regrole::text AS role, member::regrole::text,
+admin_option, inherit_option, set_option
+FROM pg_auth_members
+WHERE roleid::regrole::text = 'role6' ORDER BY 1, 2;
+
+DROP ROLE role6, role7;
+RESET citus.log_remote_commands;
+
+-- Grant role with admin option set to true and inherit option set to false
+SET citus.enable_ddl_propagation TO on;
+CREATE ROLE role9;
+CREATE ROLE role10;
+GRANT role9 TO role10 WITH ADMIN OPTION, INHERIT FALSE;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role9' ORDER BY 1, 2;
+DROP ROLE role9, role10;
+RESET citus.log_remote_commands;
+
+-- Grant role with admin option set to false and inherit option set to true
+SET citus.enable_ddl_propagation TO on;
+CREATE ROLE role11;
+CREATE ROLE role12;
+GRANT role11 TO role12;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role11' ORDER BY 1, 2;
+DROP ROLE role11, role12;
+RESET citus.log_remote_commands;
```

Comment:
Based on the comment "Grant role with admin option set to false and inherit option set to true" you wrote, this should be something like 
```suggestion
-- Grant role with admin option set to false and inherit option set to true
SET citus.enable_ddl_propagation TO on;
CREATE ROLE role11;
CREATE ROLE role12;
GRANT role11 TO role12 WITH ADMIN FALSE, INHERIT TRUE;
SELECT roleid::regrole::text AS role, member::regrole::text, admin_option FROM pg_auth_members
WHERE roleid::regrole::text = 'role11' ORDER BY 1, 2;
DROP ROLE role11, role12;
RESET citus.log_remote_commands;
```